crates/router/src/connector/bamboraapac/transformers.rs

```diff
@@ -1,234 +1,804 @@
+pub struct SubmitSingleRefundResponse {
+    submit_single_refund_result: SubmitSingleRefundResult,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "PascalCase")]
+pub struct SubmitSingleRefundResult {
+    response: RefundResponse,
 }
 
-//TODO: Fill the struct with respective fields
-#[derive(Default, Debug, Clone, Serialize, Deserialize)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "PascalCase")]
 pub struct RefundResponse {
-    id: String,
-    status: RefundStatus,
+    response_code: u8,
+    receipt: String,
+    declined_code: Option<String>,
+    declined_message: Option<String>,
 }
 
-impl TryFrom<types::RefundsResponseRouterData<api::Execute, RefundResponse>>
-    for types::RefundsRouterData<api::Execute>
+impl ForeignFrom<u8> for enums::RefundStatus {
+    fn foreign_from(item: u8) -> Self {
+        match item {
+            0 => Self::Success,
+            1 => Self::Failure,
+            _ => Self::Pending,
+        }
+    }
+}
+
+impl<F>
+    TryFrom<
+        types::ResponseRouterData<
+            F,
+            BamboraapacRefundsResponse,
+            types::RefundsData,
+            types::RefundsResponseData,
+        >,
+    > for types::RouterData<F, types::RefundsData, types::RefundsResponseData>
 {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(
-        item: types::RefundsResponseRouterData<api::Execute, RefundResponse>,
+        item: types::ResponseRouterData<
+            F,
+            BamboraapacRefundsResponse,
+            types::RefundsData,
+            types::RefundsResponseData,
+        >,
     ) -> Result<Self, Self::Error> {
+        let response_code = item
+            .response
+            .body
+            .submit_single_refund_response
+            .submit_single_refund_result
+            .response
+            .response_code;
+        let connector_refund_id = item
+            .response
+            .body
+            .submit_single_refund_response
+            .submit_single_refund_result
+            .response
+            .receipt;
+
         Ok(Self {
             response: Ok(types::RefundsResponseData {
-                connector_refund_id: item.response.id.to_string(),
-                refund_status: enums::RefundStatus::from(item.response.status),
+                connector_refund_id: connector_refund_id.to_owned(),
+                refund_status: enums::RefundStatus::foreign_from(response_code),
             }),
             ..item.data
         })
     }
 }
 
-impl TryFrom<types::RefundsResponseRouterData<api::RSync, RefundResponse>>
-    for types::RefundsRouterData<api::RSync>
+pub fn get_payment_sync_body(req: &types::PaymentsSyncRouterData) -> Result<Vec<u8>, Error> {
+    let auth_details = BamboraapacAuthType::try_from(&req.connector_auth_type)?;
+    let connector_transaction_id = req
+        .request
+        .connector_transaction_id
+        .get_connector_transaction_id()
```

Comment:
Can we use our reference id here? to recover from timeouts?

Reply:
in terms of payment, different id is generated for auth and capture, if we try to sync with connector_request_reference_id, we will have duplicate response at connnector's end, one for auth and one for capture for the same reference